### PR TITLE
feat: add expiration time

### PR DESCRIPTION
**Pull request by asafMasa**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

**Review by cl-shaharshavit-rnd — APPROVED** [review]

File: src/commands/generateToken.ts [hunk 1]
```diff
@@ -25,6 +26,9 @@ export const builder: yargs.CommandBuilder<{}, GenerateTokenArguments> = (yargs)
     demandOption: true,
   });
   yargs.option('c', { alias: 'client', type: 'string', description: 'the name of the client', demandOption: true });
+  // https://github.com/panva/jose/blob/1a3d31c467/src/lib/secs.ts
+  // https://github.com/panva/jose/blob/1a3d31c46756d4190ccc0cb9f35fa9861808ad60/src/jwt/produce.ts
```
Comment (CptSchnitz) [inline src/commands/generateToken.ts]: adding some description in addition to the links would be nice (something like "the package expects a string describing duration, for example 2 weeks")

File: src/commands/generateToken.ts [hunk 2]
```diff
@@ -25,6 +26,9 @@ export const builder: yargs.CommandBuilder<{}, GenerateTokenArguments> = (yargs)
...
+  yargs.option('t', { alias: 'ttl', type: 'string', description: '(optional) how much ttl from date.now' });
```
Comment (CptSchnitz) [inline src/commands/generateToken.ts]: 1. add to the description more information on how to specify the value.
2. no need to write optional, yargs adds it by itself to required (see image).
image

File: src/commands/generateToken.ts [hunk 3]
```diff
@@ -35,7 +39,7 @@ export const builder: yargs.CommandBuilder<{}, GenerateTokenArguments> = (yargs)
     alias: 'domains',
     type: 'array',
     array: true,
-    description: 'the map-colonies domains the client will be allowed to access (raster, 3D, etc)',
+    description: '(optional) the map-colonies domains the client will be allowed to access (raster, 3D, etc)',
```
Comment (CptSchnitz) [inline src/commands/generateToken.ts]: remove optional.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/commands/generateToken.ts [hunk 4]
```diff
@@ -26,9 +26,13 @@ export const builder: yargs.CommandBuilder<{}, GenerateTokenArguments> = (yargs)
...
   });
   yargs.option('c', { alias: 'client', type: 'string', description: 'the name of the client', demandOption: true });
+  // the jose package expects a string describing the TTL duration, for example '2h'
   // https://github.com/panva/jose/blob/1a3d31c467/src/lib/secs.ts
-  // https://github.com/panva/jose/blob/1a3d31c46756d4190ccc0cb9f35fa9861808ad60/src/jwt/produce.ts
-  yargs.option('t', { alias: 'ttl', type: 'string', description: '(optional) how much ttl from date.now' });
+  yargs.option('t', {
+    alias: 'ttl',
+    type: 'string',
+    description: 'how many TTL from date.now, the package will take the supplied time and add it to the time the token was generated',
```
Comment (CptSchnitz) [inline src/commands/generateToken.ts]: add to the description like you added in the comment "for example '2h'"

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**